### PR TITLE
BuddyPress Unit Tests: review our tests

### PR DESCRIPTION
**Pull request by renatonascalves**
Trac ticket: https://buddypress.trac.wordpress.org/ticket/9081

---
**This Pull Request is for code review only. Please keep all other discussion in the BuddyPress Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the WordPress Core Handbook for more details.**

### CODE REVIEW

**Comment by imath** [comment]
Amazing @renatonascalves !!! I'll try to review it asap. Do you plan to include it into the 14.0 milestone? I'm asking because it's currently set to `up next` on Trac.

**Comment by renatonascalves** [comment]
@imath If it's approved, sure. We can also delay, no hurry.

**Comment by dcavins** [comment]
Wow! Does this win the award for most extensive patch ever? :)

**Comment by renatonascalves** [comment]
Considering the `loc`, [this one](https://github.com/buddypress/BP-REST/pull/508) is bigger.

File: tests/phpunit/testcases/core/avatars.php [hunk 1]
```diff
@@ -8,14 +8,10 @@ class BP_Tests_Avatars extends BP_UnitTestCase {
...
 
-	public function set_up() {
-		parent::set_up();
-	}
-
 	private function clean_existing_avatars( $type = 'user' ) {
 		if ( 'user' === $type ) {
 			$avatar_dir = 'avatars';
-		} elseif ( 'group' === $object ) {
+		} elseif ( 'group' === $type ) {
```
Comment (imath) [inline tests/phpunit/testcases/core/avatars.php:14]: Good catch!!

File: tests/phpunit/includes/testcase-emails.php [hunk 2]
```diff
@@ -14,22 +14,4 @@ public static function set_up_before_class() {
...
-		) );
-
-		if ( $emails ) {
-			foreach ( $emails as $email_id ) {
-				wp_delete_post( $email_id, true );
-			}
-		}
-
-		parent::tear_down_after_class();
-	}
```
Comment (imath) [inline tests/phpunit/includes/testcase-emails.php:34]: I just need a confirmation: these emails are automatically removed at the next time we launch the tests, that's why you think it's not needed to removed them here?

**Review by imath — APPROVED** [review]
Hi @renatonascalves 

Thanks for reminding me about it. Enormous work 💪 , thanks a lot for it. I just need a confirmation about the BP Emails removal.

Feel free to SVN commit it!

BTW, as we're talking about unit tests, your improvements about testing cache softwares as well as without them seem to be fine, I believe like you initially suggested: we can now remove the `.github/workflows/unit-tests.yml` action.

File: tests/phpunit/includes/testcase-emails.php (hunk 2 above)
Comment (renatonascalves) [reply tests/phpunit/includes/testcase-emails.php:34]: @imath That's correct. All posts are removed at tear_down_after_class. See https://github.com/WordPress/wordpress-develop/blob/859263db0657a7f536fdeefe60a1225ceb6fd4f7/tests/phpunit/includes/functions.php#L110

**Pull request closed without merging**